model/src/main/kotlin/licenses/License.kt

```diff
@@ -49,7 +50,8 @@ data class License(
     val includeSourceCodeOfferInNoticeFile: Boolean
 ) {
     init {
-        require(id.isNotEmpty()) { "The identifier must not be empty." }
-        // TODO: check if the id is a valid SPDX identifier.
+        require(SpdxExpression.parse(id, SpdxExpression.Strictness.ALLOW_ANY).licenses().size == 1) {
```

Comment:
Don't you want to explicitly catch `SpdxException` in this case? So how about this slightly longer but more speaking check instead:

        val expression = try {
            SpdxExpression.parse(id)
        } catch (e: SpdxException) {
            null
        }

        require(expression != null && (expression !is SpdxCompoundExpression || expression.operator == SpdxOperator.WITH)) {
        //...

Reply:
> Don't you want to explicitly catch `SpdxException` in this case?

I thought about that too. I decided against it because in order to get a meaningful error message create from the 'require' I would need to use two require blocks for that or a case distinction for the exception while just letting the SpdxExpression propagate already gives a meaningful error message. I still prefer this imo more leightweight approach.

> So how about this slightly longer but more speaking check instead:

I choose the implemention in this way because it seems simpler and easier to understand and most importantly need less understanding of the internals of the spdx-utils module.
I prefer keeping as is.

What do you think ?

Reply:
I'd be fine with keeping your version if we could do two things:

1. Omit the `SpdxExpression.Strictness.ALLOW_ANY` default argument. I guess you wanted to be explicit for clarity, but if we're always being explicit about this parameter, we don't need a default argument for the function at all.

2. Add a code comment that explains what kind of `id`s are acceptable here (e.g. "We expect id to either be a compound expression with an exception, or no compound expression at all") and / or add such a remark to the [docs for id](https://github.com/heremaps/oss-review-toolkit/pull/1925/files#diff-c4933700a89f0e2b2bfcaf68c18e3d05R33).

Reply:
>     1. I guess you wanted to be explicit for clarity

In this case actually I wanted to minimize the risk of `changes in spdx-utils` breaking the check also a reason why I used `licenses().size == 1`, but anyhow I'm perfectly fine with your proposal. So thanks, will do...!

Reply:
Yeah, when changing default arguments you should always be aware that this might break a caller's experience.